src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -121,6 +119,12 @@ public PsiElement resolve() {
     return resolve(true);
   }
 
+  public boolean resolveIsExtension() {
+    // @TODO: DIRTY HACK! to avoid rewriting all the code!
```

Comment:
I have done it with a ThreadLocal temporal solution, as I think I don't understand completely what you mean. We can change it later.